### PR TITLE
Accept raw Mongo Urls

### PR DESCRIPTION
**Pull request by andresespinosapc**
I allow to pass a raw mongo url and pass it directly to `MongoClient`, because you don't support `mongo+srv` protocol, which I needed to use. For this purpose, I had to remove `as_mongo_str` on `mongoexp.py` at line 1251, but I don't see any disadvantages of doing that, besides you have to update the documentation.

### CODE REVIEW

**Comment by andresespinosapc** [comment]
It's weird, because the tests on the branch master are not passing in my computer neither :/ I don't know what is failing, help please

File: hyperopt/mongoexp.py [hunk 1]
```diff
@@ -200,13 +200,13 @@ def parse_url(url, pwfile=None):
     logger.info('HOSTNAME %s' % tmp.hostname)
     logger.info('PORT %s' % tmp.port)
     logger.info('PATH %s' % tmp.path)
-    
+
```
Comment (michaelmior) [inline hyperopt/mongoexp.py]: Please keep whitespace changes out of PRs since it just pollutes the diff.

File: hyperopt/mongoexp.py [hunk 2]
```diff
@@ -223,7 +223,8 @@ def parse_url(url, pwfile=None):
     else:
         password = tmp.password
     logger.info('PASS %s' % password)
-    port = int(float(tmp.port))  # port has to be casted explicitly here.
+    # If no port given, set it to de defaul (27017)
+    port = int(float(tmp.port)) if tmp.port else 27017  # port has to be casted explicitly here.
```
Comment (michaelmior) [inline hyperopt/mongoexp.py]: This seems to be unrelated to your proposed change

**Comment by michaelmior** [comment]
This removes the ability to pass in a connection string without a protocol. Instead, I would suggest changing `as_mongo_str` to check for any protocol, adding `mongo://` as the default if none exists. This will allow `mongo+srv` to be used without breaking existing functionality.

**Comment by michaelmior** [comment]
Is there now a way to use the `mongo+srv` protocol?
